### ISSUE TITLE
document viewer

### ISSUE THREAD

**Issue by chensce** [frappe/drive#7]
There are some plan about viewer?

**Comment by agritheory** [comment]
@chensce Not all MIME types are currently handled, but PDFs and images look good to me in the viewer. What types were you looking for?
![Untitled](https://user-images.githubusercontent.com/18033666/191989182-f22ca260-04de-4490-b263-74afb889d0de.png)

**Comment by chensce** [comment]
ppt，docs，excel

**Comment by chensce** [comment]
mp4，mp3

**Comment by chensce** [comment]
thank you！😄

**Comment by krantheman** [comment]
@chensce there are no immediate plans, but do we do plan on adding these some time in the future.

**Comment by agritheory** [comment]
@krantheman Akash, I'd like to propose a plan to prioritize preview elements and a pattern to add custom viewers (and possibly MIMEType icons). There's a large group of file types that makes sense to incorporate viewers for, but there will be others that are going to be industry specific. I think that an API that allows you to extend Drive by offering a custom preview could be really valuable. I can rough out a design for this, but want to understand your level of interest first.

**Comment by krantheman** [comment]
@agritheory sure, what file types are you thinking?

**Comment by agritheory** [comment]
@krantheman What I am thinking is going for the most surface area (file types by percentage) while also keeping dependencies to a minimum. 
First round, I'd like to propose previews for five MIME types: markdown, CSV, XLS, XLSX and ODT. Markdown can be handled with existing dependencies, The others are handled by the same package: https://www.npmjs.com/package/excel-viewer
Second round seems like DOC, DOCX and the Open Office / Libre office format, but there isn't a clear shortcut there like there is with the spreadsheet category: https://www.npmjs.com/search?q=docx
Slides / PowerPoint after that, I haven't looked into dependencies for these. 
Then video and audio, where there are lots of options but it makes sense to use the same deps as the Frappe framework's A/V stack.
We don't need to solve for all files, we need to solve for the most encountered ones. There's always going to be some file type that is specific and common to one use case and would never be used in another. A manufacturer might often need to share drawings (DXF is one of the most common) but that's something a software development business would like never need to deal with. Providing an API where Drive can be extended to support arbitrary previews would be an excellent feature, but getting some of the most common previews built in should be the first step.

**Comment by krantheman** [comment]
Hmm, sounds good to me. Are you willing to contribute to the development of this?

**Comment by agritheory** [comment]
@krantheman Yes. I'll try my hand at the markdown previewer first and then the spreadsheets. That will give enough of a body of work to know if there should be a design change.

**Comment by krantheman** [comment]
Cool! Also, I think I should let you know that there are plans to implement the frappe-ui text editor for simple text editing. I haven't dabbled enough to know about its usage though.

**Comment by RUTU143** [comment]
> @krantheman Yes. I'll try my hand at the markdown previewer first and then the spreadsheets. That will give enough of a body of work to know if there should be a design change.

my frappe drive will not open why?
![Screenshot from 2023-01-28 11-43-12](https://user-images.githubusercontent.com/106966297/215250542-04749298-e594-45c2-857c-6da5854185aa.png)
![Screenshot from 2023-01-28 11-42-58](https://user-images.githubusercontent.com/106966297/215250544-ff21361a-d752-44ce-8829-71c659d8889d.png)
